spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -407,7 +413,11 @@ public final boolean send(Message<?> message, long timeout) {
 			if (this.datatypes.length > 0) {
 				message = this.convertPayloadIfNecessary(message);
 			}
-			if (interceptors.getInterceptors().size() > 0) {
+			if (logger.isDebugEnabled()) {
+				logger.debug("preSend on channel '" + this + "', message: " + message);
+			}
+			boolean hasInterceptors = interceptors.getInterceptors().size() > 0;
```

Comment:
The local var has 2 purposes:

1. Avoid a second invocation of `getInterceptors().size()`
2. Only call postSend() if we called preSend() (the interceptors could change).

Yes, we should optimize the other uses; will push another commit.

When this PR is merged I'll open a JIRA against SF.